addons/binding/org.openhab.binding.irtrans/src/main/java/org/openhab/binding/irtrans/handler/EthernetBridgeHandler.java

```diff
@@ -0,0 +1,925 @@
+     **/
+    protected final Collection<IrCommand> irCommands = new HashSet<IrCommand>();
+
+    public EthernetBridgeHandler(Bridge bridge) {
+        super(bridge);
+        // Nothing to do here
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing IRtrans Ethernet handler.");
+
+        // register ourselves as a Transceiver Status Listener
+        registerTransceiverStatusListener(this);
+
+        try {
+            selector = Selector.open();
+        } catch (IOException e) {
+            logger.error("An exception occurred while registering the selector: '{}'", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR, e.getMessage());
+        }
+
+        if (selector != null) {
+            if (getConfig().get(IP_ADDRESS) != null && getConfig().get(PORT_NUMBER) != null) {
+                onUpdate();
+            } else {
+                logger.warn("Cannot connect to IRtrans Ethernet device. IP address or port number not set.");
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "IP address or port number not set.");
+            }
+
+            if (getConfig().get(IS_LISTENER) != null) {
+                configureListener((String) getConfig().get(LISTENER_PORT));
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposed.");
+
+        unregisterTransceiverStatusListener(this);
+
+        disconnect(socketChannel);
+
+        try {
+            if (selector != null) {
+                selector.close();
+            }
+        } catch (IOException e) {
+            logger.error("An exception occurred while closing the selector: '{}'", e.getMessage());
+        }
+
+        if (pollingJob != null && !pollingJob.isCancelled()) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+    }
+
+    public boolean registerTransceiverStatusListener(TransceiverStatusListener transceiverStatusListener) {
+        Objects.requireNonNull(transceiverStatusListener, "It's not allowed to pass a null BlasterStatusListener.");
+
+        boolean result = transceiverStatusListeners.add(transceiverStatusListener);
+        if (result) {
+            onUpdate();
+        }
+        return result;
+    }
+
+    public boolean unregisterTransceiverStatusListener(TransceiverStatusListener transceiverStatusListener) {
+        boolean result = transceiverStatusListeners.remove(transceiverStatusListener);
+        if (result) {
+            onUpdate();
+        }
+        return result;
+    }
+
+    private synchronized void onUpdate() {
+        if (pollingJob == null || pollingJob.isCancelled()) {
+            pollingJob = scheduler.scheduleWithFixedDelay(pollingRunnable, 0,
+                    ((BigDecimal) getConfig().get(REFRESH_INTERVAL)).intValue(), TimeUnit.MILLISECONDS);
```

Comment:
I think we had this discussion at some other place before: I doubt that it makes sense to use the scheduler for jobs that are scheduled every 50ms. Especially if these jobs then do a lot of `Thread.sleep`s for much more than 50ms - effectively, those jobs will more or less block a complete thread from the shared thread pool.
If you need to permanently listen for updates, wouldn't a dedicated thread make more sense?

Reply:
See also https://github.com/openhab/openhab2-addons/pull/2284#discussion_r155348106